query/src/main/scala/filodb/query/exec/PlanDispatcher.scala

```diff
@@ -26,10 +28,11 @@ trait PlanDispatcher extends java.io.Serializable {
   */
 case class ActorPlanDispatcher(target: ActorRef) extends PlanDispatcher {
 
-  def dispatch(plan: ExecPlan)
-              (implicit sched: Scheduler,
-               timeout: FiniteDuration): Task[QueryResponse] = {
-    implicit val _ = Timeout(timeout)
+  def dispatch(plan: ExecPlan)(implicit sched: Scheduler): Task[QueryResponse] = {
+
+    val queryTime = (System.currentTimeMillis() - plan.queryContext.submitTime) / 1000
+    if (queryTime >= plan.queryContext.queryTimeoutSecs) throw QueryTimeoutException(queryTime, this.getClass.getName)
+    implicit val _ = Timeout(FiniteDuration(plan.queryContext.queryTimeoutSecs - queryTime, TimeUnit.SECONDS))
```

Comment:
dont create implicit here for single use. Just pass it in explicitly. Also you need to convert to millis.